frontend/app/routes/_public+/apply+/$id+/personal-information.tsx

```diff
@@ -110,16 +110,22 @@ export async function action({ request, params }: ActionFunctionArgs) {
       }
 
       if (val.copyMailingAddress !== 'on') {
+        const homeAddressErrorMessages = [
+          { fieldName: 'homeAddress', errorMessage: 'empty-address' },
+          { fieldName: 'homeCountry', errorMessage: 'empty-country' },
+          { fieldName: 'homeCity', errorMessage: 'empty-city' },
+          { fieldName: 'homePostalCode', errorMessage: 'empty-postal-code' },
+        ];
         const homeAddessFields: (keyof typeof val)[] = ['homeAddress', 'homeCountry', 'homeCity', 'homePostalCode'];
         homeAddessFields.forEach((field) => {
           if (isEmpty(val[field])) {
-            ctx.addIssue({ code: z.ZodIssueCode.custom, message: 'empty-field', path: [field] });
+            ctx.addIssue({ code: z.ZodIssueCode.custom, message: homeAddressErrorMessages.find((fld) => fld.fieldName === field)?.errorMessage, path: [field] });
           }
         });
```

Comment:
**Suggestion:** This can be further simplified (IMO) to use one variable to hold the field names and error messages:

```typescript
const homeAddessFields: { fieldName: keyof typeof val; errorMessage: string }[] = [
  { fieldName: 'homeAddress', errorMessage: 'empty-address' },
  { fieldName: 'homeCountry', errorMessage: 'empty-country' },
  { fieldName: 'homeCity', errorMessage: 'empty-city' },
  { fieldName: 'homePostalCode', errorMessage: 'empty-postal-code' }
];

homeAddessFields.forEach(({ fieldName, errorMessage }) => {
  if (isEmpty(val[fieldName])) {
    ctx.addIssue({ code: z.ZodIssueCode.custom, message: errorMessage, path: [fieldName] });
  }
});
```

Reply:
fixed